src/net/p2p.rs

```diff
@@ -3446,23 +3509,45 @@ impl PeerNetwork {
                         }
                     }
                 }
-                MempoolSyncState::RecvResponse(ref event_id) => {
+                MempoolSyncState::RecvResponse(ref url, ref addr, ref event_id) => {
                     match self.mempool_sync_recv_response(*event_id)? {
-                        (true, Some(txs)) => {
+                        (true, next_page_id_opt, Some(txs)) => {
+                            debug!(
+                                "{:?}: Mempool sync received {} transactions; next page is {:?}",
+                                &self.local_peer,
+                                txs.len(),
+                                &next_page_id_opt
+                            );
+
                             // done! got data
-                            self.mempool_sync_reset();
-                            return Ok((true, Some(txs)));
+                            let ret = match next_page_id_opt {
+                                Some(next_page_id) => {
+                                    // get the next page
+                                    self.mempool_state = MempoolSyncState::SendQuery(
+                                        url.clone(),
+                                        addr.clone(),
+                                        next_page_id,
+                                    );
+                                    false
+                                }
```

Comment:
This branch appears to be uncovered as well.

Reply:
That's definitely getting exercised in `test_mempool_sync_2_peers_paginated`.